src/common/hist_util.cc

```diff
@@ -248,17 +247,19 @@ void GHistIndexMatrix::Init(DMatrix* p_fmat, int max_num_bins) {
   size_t prev_sum = 0;
 
   for (const auto &batch : p_fmat->GetRowBatches()) {
+    auto batch_size = static_cast<omp_ulong>(batch.Size());
+
     MemStackAllocator<size_t, 128> partial_sums(nthread);
     size_t* p_part = partial_sums.Get();
 
-    size_t block_size =  batch.Size() / nthread;
```

Comment:
Why not use batch.Size()? Is this expected to change?

Reply:
since it is used in multiple places, i only invoke this once and use that variable elsewhere.

Reply:
`batch.Size()` is the "most true" source of this information so you should use it directly if there is no performance penalty. This may seem pedantic but adding layers between effects readability and creates assumptions. For example that batch.Size() has not changed since you first called it - in this case maybe a reasonable assumption, but in general this as a reason not to do this.

Hope this makes sense.

Reply:
i do not want to make assumptions on performance implications without profiling first. hence, this wasn't done due to performance reasons.

i'm trying to understand how using a variable named `batch_size` as opposed to using `batch.Size()` impacts readability, given that this is used in multiple places in the code block. by using the variable here, we are also conveying the intent that a new value isn't used at the call site and that a constant value computed at the beginning of the block is used.  further, this also localizes the means to compute the batch size in a single place, such that *if* the way to compute the retrieve the batch size changes later, the change is limited to a single place.